src/net_processing.cpp

```diff
@@ -1452,15 +1456,11 @@ bool PeerManagerImpl::FetchBlock(NodeId id, const uint256& hash, const CBlockInd
         return true;
     });
 
-    if (success) {
-        LogPrint(BCLog::NET, "Requesting block %s from peer=%d\n",
-                 hash.ToString(), id);
-    } else {
-        RemoveBlockRequest(hash);
-        LogPrint(BCLog::NET, "Failed to request block %s from peer=%d\n",
+    if (!success) return "Node not fully connected";
```

Comment:
Maybe add a test assertion for this in the functional test.

Reply:
I have no idea how to produce that scenario; IIUC there's only a very brief interval between when a peer id is assigned and the handshake is done.

Reply:
```suggestion
    if (!success) return "Peer not fully connected";
```